canonical/canonical.com
# Implement the navigation and reduced navigation

## Done

- Implement the [navigation](https://vanillaframework.io/docs/examples/patterns/navigation/search-dark?theme=light) and [reduced navigation](https://vanillaframework.io/docs/examples/patterns/navigation/reduced?theme=light) HTML patterns
- Add Python logic to handle secondary navigation
- Add JS to handle the states of search and secondary navigation (dropdowns, mobile menu don't work). Most is boiler plate JS copied from Vanilla, u.com and the other c.com navigation

## QA

- Go https://canonical-com-1290.demos.haus/ to and see the implementation of the regular navigation matches the [vanilla navigation](https://vanillaframework.io/docs/examples/patterns/navigation/search-dark?theme=light)
- Go to https://canonical-com-1290.demos.haus/data and see the implementation of the reduced navigation matches the [reduced navigation](https://vanillaframework.io/docs/examples/patterns/navigation/reduced?theme=light)
- Check the search opens and closes as expected on the normal navigation, reduced navigation and mobile (for both)
- Check that on mobile, the secondary navigation dropdown functions properly

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-12668

## Screenshots
Regular navigation:
![image](https://github.com/canonical/canonical.com/assets/58276363/6e479c78-68a4-4535-b374-6ead23ea87c0)
Reduced navigation:
![image](https://github.com/canonical/canonical.com/assets/58276363/a7c6dbf8-ebce-485e-89b9-8a92acd93a57)
Regular navigation (search open):
![image](https://github.com/canonical/canonical.com/assets/58276363/139b9426-bebc-44fd-b788-0b6092098257)
Reduced navigation (search open):
![image](https://github.com/canonical/canonical.com/assets/58276363/8426e732-46e2-4bf0-88d5-070051a7ad3b)
